src/main/java/frc/robot/commands/AwaitVisionResult.java

```diff
@@ -0,0 +1,30 @@
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.Vision;
+import frc.robot.subsystems.Vision.VisionMode;
+
+public class AwaitVisionResult extends CommandBase {
+    private Vision m_vision;
+    private VisionMode m_mode;
+    public AwaitVisionResult(Vision vision, VisionMode mode) {
+        m_vision = vision;
+        m_mode = mode;
+    }
+    @Override
+    public void initialize() {
+        m_vision.setMode(m_mode);
+    }
+    @Override
+    public boolean isFinished() {
+        return m_vision.getStatus();
+    }
+    @Override
+    public void execute() {
+    }
+    @Override
+    public void end(boolean interrupted) {
+        if(interrupted) m_vision.setMode(Vision.DEFAULT_MODE);
```

Comment:
Since we didn't define a required subsystem for this command, it will neve be interrupted.

Reply:
In the autos the command is wrapped with withTimeout so that it will be interrupted if it takes too long